src/block/block_compact.c

```diff
@@ -419,9 +419,9 @@ __wt_block_compact_progress(WT_SESSION_IMPL *session, WT_BLOCK *block, u_int *ms
     __wt_epoch(session, &cur_time);
 
     /* Log one progress message every twenty seconds. */
-    time_diff = WT_TIMEDIFF_SEC(cur_time, session->compact->begin);
-    if (time_diff / WT_PROGRESS_MSG_PERIOD > *msg_countp) {
-        ++*msg_countp;
+    time_diff = WT_TIMEDIFF_SEC(cur_time, session->compact->last_progress);
+    if (time_diff > WT_PROGRESS_MSG_PERIOD) {
+        __wt_epoch(session, &session->compact->last_progress);
```

Comment:
Minor optimization: can we just use `cur_time` here instead of reading the clock again?

Reply:
Yes, that makes more sense